File: app/src/main/java/c/m/storyapp/login/data/mapper/LoginMapper.kt
package c.m.storyapp.login.data.mapper

import c.m.storyapp.login.data.remote.dto.LoginResponseDTO
import c.m.storyapp.login.data.remote.dto.LoginResultDTO
import c.m.storyapp.login.domain.model.LoginResponse
import c.m.storyapp.login.domain.model.LoginResult

fun LoginResultDTO.toLoginResult(): LoginResult {
    return LoginResult(
        userId = userId,
        name = name,
        token = token,
    )
}

fun LoginResponseDTO.toLoginResponse(): LoginResponse {
    return LoginResponse(
        error = error,
        message = message,
        loginResult = loginResult.toLoginResult()
    )
}

File: lib/login/data/mapper/login_mapper.dart
import 'package:story_app/login/data/remote/dto/login_response_dto.dart';
import 'package:story_app/login/data/remote/dto/login_result_dto.dart';
import 'package:story_app/login/domain/model/login_response.dart';
import 'package:story_app/login/domain/model/login_result.dart';

extension LoginResultDTOExtension on LoginResultDTO {
  LoginResult toLoginResult() => LoginResult(
        userId: userId,
        name: name,
        token: token,
      );
}

extension LoginResponseDTOExtension on LoginResponseDTO {
  LoginResponse toLoginResponse() => LoginResponse(
      error: error,
      message: message,
      loginResult: loginResultDTO?.toLoginResult());
}
